### ISSUE TITLE
Codepush on React Native 0.69 with Hermes is broken

### ISSUE THREAD

**Issue by zolbooo** [microsoft/appcenter-cli#1987]
### **Description**
React Native 0.69 now uses [bundled Hermes engine](https://reactnative.dev/docs/hermes#bundled-hermes). Currently, codepush relies on `hermes-engine` package. For example, CLI uses Hermes engine 0.11.0 for RN 0.68 with bytecode version 84, but current bytecode version (at the time of writing this issue is 85).

### **Repro Steps**
1. Use React Native 0.69 app, install it on device
2. Create new deployment `appcenter codepush release-react ...`
3. Open app and wait for OTA update to be downloaded
4. Restart app
5. App crashes and OTA is rolled back

### **Expected behavior**
App loaded OTA update should work correctly.

### **Details**
Codepush should use `hermesc` bundled with current version of React Native.

1. Is there a particular beacon/command that you are having trouble with?
    - `appcenter codepush release-react`
2. Which version of the App Center CLI are you using?
    - 2.10.10
6. Which OS version did you experience the issue on?
    - macOS Monterey 12.3.1
7. What command line/shell are you using?
    - zsh 5.8
8. What version of Node.js and NPM/Yarn are you using?
    - node v16.13.2, npm 8.13.1
9. Example crash stacktrace on Android:
```
com.facebook.jni.CppException: Compiling JS failed: Wrong bytecode version. Expected 85 but got 84 Buffer size 3836525 starts with: c61fbc03c103191f54000000ab331809 and has protection mode(s): r--p
        at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
        at android.os.Handler.handleCallback(Handler.java:907)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(Unknown Source:0)
        at android.os.Looper.loop(Looper.java:216)
        at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(Unknown Source:37)
        at java.lang.Thread.run(Thread.java:784)
```

UPD: Related issue - https://github.com/microsoft/react-native-code-push/issues/2287

**Comment by aleksandr-dorofeev** [comment]
Hello @zolbooo, thanks for contacting us.
We are currently reviewing the [PR](https://github.com/microsoft/appcenter-cli/pull/1325/) with the fix. That contains this fix as well. You can follow the progress in it.

**Comment by zolbooo** [comment]
Hi, @aleksandr-dorofeev.
I think this PR could work on projects with custom setup, but it would still fail on RN 0.69+. I have created my own PR and I'm testing it in some personal projects.

**Comment by DmitriyKirakosyan** [comment]
Hi @zolbooo , you are right, sorry for the confusion.
I left a comment in your PR, please have a look.

**Comment by Fagner3g** [comment]
problem persists in 0.69.1 :/

**Comment by ahmadzraiq** [comment]
@aleksandr-dorofeev How can I run this fix on react-native-code-push

**Comment by aleksandr-dorofeev** [comment]
Hi @ahmadzraiq
Aither wait for the next release or build `appcenter-cli` locally.

**Comment by geekyoperand** [comment]
Hi @aleksandr-dorofeev, Does code push supports react native version 0.68.1. Because as per the documentation only these versions are mentioned

![image](https://user-images.githubusercontent.com/52500177/181120977-a8642b3a-0ebb-4954-afc1-af75a2df5934.png)

**Comment by DmitriyKirakosyan** [comment]
@geekyoperand you are asking about the [react-native-code-push](https://github.com/microsoft/react-native-code-push) package, right? It supports all new versions. But we will probably need to update the table, thanks for reminding.
Please file an issue in the corresponding repo next time, as the question is not related to the appcenter-cli.

**Comment by DmitriyKirakosyan** [comment]
As for this issue, sorry for the confusion, it was closed unintentionally. Reopening since the fix is not released yet.
cc @ahmadzraiq

**Comment by zolbooo** [comment]
> @aleksandr-dorofeev How can I run this fix on react-native-code-push

You can use `patch-package`, add `patches/appcenter-cli+2.10.10.patch`, here is one that I use:
```
diff --git a/node_modules/appcenter-cli/dist/commands/codepush/lib/react-native-utils.js b/node_modules/appcenter-cli/dist/commands/codepush/lib/react-native-utils.js
index 1e67a28..d1f01c9 100644
--- a/node_modules/appcenter-cli/dist/commands/codepush/lib/react-native-utils.js
+++ b/node_modules/appcenter-cli/dist/commands/codepush/lib/react-native-utils.js
@@ -390,7 +390,7 @@ function getAndroidHermesEnabled(gradleFile) {
         throw new Error(`Unable to parse the "${buildGradlePath}" file. Please ensure it is a well-formed Gradle file.`);
     })
         .then((buildGradle) => {
-        return Array.from(buildGradle["project.ext.react"] || []).includes("enableHermes: true");
+        return Array.from(buildGradle["project.ext.react"] || []).some((line) => /^enableHermes\s{0,}:\s{0,}true/.test(line));
     });
 }
 exports.getAndroidHermesEnabled = getAndroidHermesEnabled;
@@ -443,6 +443,11 @@ function getHermesCommand() {
             return false;
         }
     };
+    // Hermes is bundled with react-native since 0.69
+    const bundledHermesEngine = path.join("node_modules", "react-native", "sdks", "hermesc", getHermesOSBin(), getHermesOSExe());
+    if (fileExists(bundledHermesEngine)) {
+        return bundledHermesEngine;
+    }
     // assume if hermes-engine exists it should be used instead of hermesvm
     const hermesEngine = path.join("node_modules", "hermes-engine", getHermesOSBin(), getHermesOSExe());
     if (fileExists(hermesEngine)) {

```

**Comment by zolbooo** [comment]
@AnatolyPristensky do you have any ETAs when this fix will be released?

**Comment by afrakhan123** [comment]
if I install the cli from master to fix the hermes problem i get this error 
Cannot find module '../src/util/commandline'
@zolbooo pls can you help?

**Comment by eshantbist** [comment]
@afrakhan123 

You can use what @zolbooo has suggested.

Install appcenter-cli in your project and use the suggested patch.

**Comment by AnatolyPristensky** [comment]
Hello @zolbooo,
The fix is released and available in [v2.11.0](https://github.com/microsoft/appcenter-cli/releases/tag/v2.11.0)
